src/proposals/20241114/SparkEthereum_20241114.t.sol

```diff
@@ -0,0 +1,428 @@
+    }
+
+    function testWETHInterestRateUpdate() public {
+        ReserveConfig[] memory allConfigsBefore = createConfigurationSnapshot('', pool);
+
+        ReserveConfig memory wethConfigBefore = _findReserveConfigBySymbol(allConfigsBefore, 'WETH');
+
+        assertEq(wethConfigBefore.interestRateStrategy, OLD_WETH_INTEREST_RATE_STRATEGY);
+
+        uint256 expectedOldSlope1 = 0.028991774151952311e27;
+        InterestStrategyValues memory values = InterestStrategyValues({
+            addressesProvider:             address(poolAddressesProvider),
+            optimalUsageRatio:             0.9e27,
+            optimalStableToTotalDebtRatio: 0,
+            baseStableBorrowRate:          expectedOldSlope1,
+            stableRateSlope1:              0,
+            stableRateSlope2:              0,
+            baseVariableBorrowRate:        0,
+            variableRateSlope1:            expectedOldSlope1,
+            variableRateSlope2:            1.2e27
+        });
+        _validateInterestRateStrategy(
+            wethConfigBefore.interestRateStrategy,
+            OLD_WETH_INTEREST_RATE_STRATEGY,
+            values
+        );
+
+        executePayload(payload);
+
+        ReserveConfig[] memory allConfigsAfter = createConfigurationSnapshot('', pool);
+
+        ReserveConfig memory wethConfigAfter = _findReserveConfigBySymbol(allConfigsAfter, 'WETH');
+
+        uint256 expectedSlope1 = expectedOldSlope1 - 0.005e27;
+        assertEq(expectedSlope1, 0.023991774151952311e27);
+
+        values.baseStableBorrowRate = expectedSlope1;
+        values.variableRateSlope1   = expectedSlope1;
+        _validateInterestRateStrategy(
+            wethConfigAfter.interestRateStrategy,
+            NEW_WETH_INTEREST_RATE_STRATEGY,
+            values
+        );
+    }
+
+    function testWSTETHBorrowCapUpdate() public {
+        _assertBorrowCapConfig({
+            asset:            Ethereum.WSTETH,
+            max:              3_000,
+            gap:              100,
+            increaseCooldown: 12 hours
+        });
+
+        executePayload(payload);
+
+        _assertBorrowCapConfig({
+            asset:            Ethereum.WSTETH,
+            max:              100_000,
+            gap:              5_000,
+            increaseCooldown: 12 hours
+        });
+    }
+
+    function testALMControllerDeployment() public {
+        // Copied from the init library, but no harm checking this here
+        IALMProxy         almProxy   = IALMProxy(Ethereum.ALM_PROXY);
+        IRateLimits       rateLimits = IRateLimits(Ethereum.ALM_RATE_LIMITS);
+        MainnetController controller = MainnetController(Ethereum.ALM_CONTROLLER);
+
+        assertEq(almProxy.hasRole(0x0, Ethereum.SPARK_PROXY),   true, "incorrect-admin-almProxy");
+        assertEq(rateLimits.hasRole(0x0, Ethereum.SPARK_PROXY), true, "incorrect-admin-rateLimits");
+        assertEq(controller.hasRole(0x0, Ethereum.SPARK_PROXY), true, "incorrect-admin-controller");
+
```

Comment:
Lets add this:


```solidity
        assertEq(almProxy.hasRole(0x0, DEPLOYER), false,   "incorrect-admin-almProxy");
        assertEq(rateLimits.hasRole(0x0, DEPLOYER), false, "incorrect-admin-rateLimits");
        assertEq(controller.hasRole(0x0, DEPLOYER), false, "incorrect-admin-controller");
```

with `address internal constant DEPLOYER = 0x6F3066538A648b9CFad0679DF0a7e40882A23AA4;` at the top

Reply:
done